include/fmt/format.h

```diff
@@ -1213,7 +1213,7 @@ FMT_CONSTEXPR auto format_decimal(OutputIt out, UInt value, int num_digits)
     return out;
   }
   // Buffer is large enough to hold all digits (digits10 + 1).
-  char buffer[digits10<UInt>() + 1];
+  char buffer[digits10<UInt>() + 1]{};
   if (is_constant_evaluated()) fill_n(buffer, sizeof(buffer), '\0');
```

Comment:
In case the unconditional `buffer` initialization is acceptable the `fill_n` invocations can be dropped.